### ISSUE TITLE
mypy fails in the CI with Python 3.10

### ISSUE THREAD

**Issue by dalonsoa** [ImperialCollegeLondon/pycsvy#16]
Odd error when running mypy in the CI system with Python 3.10 - for the three platforms. Might it be a glitch?
```
=================================== FAILURES ===================================
_________________________________ test session _________________________________
mypy exited with status 2.

---------- coverage: platform darwin, python 3.10.7-final-0 ----------
Coverage HTML written to dir reports/coverage/html
Coverage XML written to file reports/coverage/coverage.xml

===================================== mypy =====================================
/Users/runner/Library/Caches/pypoetry/virtualenvs/pycsvy-NphvK76O-py3.10/lib/python3.10/site-packages/numpy/__init__.pyi:6[42](https://github.com/ImperialCollegeLondon/pycsvy/actions/runs/3268256575/jobs/5374463686#step:6:43): error: Positional-only parameters are only supported in Python 3.8 and greater
Found 1 error in 1 file (errors prevented further checking)
=========================== short test summary info ============================
FAILED csvy/__init__.py::mypy-status
======================== 1 failed, 25 passed in 17.13s =========================
Error: Process completed with exit code 1.
```

**Comment by dalonsoa** [comment]
@all-contributors please add @dalonsoa for infra, ideas, maintenance, test, bug and code

**Comment by AdrianDAlessandro** [comment]
Appears to be a bug in mypy and [was fixed in version 0.981](https://github.com/python/mypy/issues/13627#issuecomment-1240907370)
